### DISCUSSION TITLE
Modify root crontab

### DISCUSSION THREAD

**Discussion by launchdaemon** [Security-Onion-Solutions/securityonion#3165] [Unsupported Versions]
I have a standalone server on version 2.3.21 installed from github on Ubuntu server 1804.  During setup I selected suricata rather than zeek.

I am trying to configure email and have installed and successfully tested `nullmailer`. However I now receive emails every 10 minutes from cron that it can't find the container `so-zeek` when the `packetloss.sh` script is run.

What is the supported/ required way to stop this running?  I assume there is no reason for it to if zeek is not configured?

The crontab states it is managed by salt and not to edit it. I can see that `packetloss.sh` is specified in `/opt/so/saltstack/default/salt/zeek/init.sls` but there is also a `/opt/so/saltstack/default/pillar/zeek/init.sls` with a line `- misc/capture-loss` that I assumed may be related?

I tried commenting out this section:

```
#zeekpacketlosscron:
#  cron.present:
#    - name: /usr/local/bin/packetloss.sh
#    - user: root
#    - minute: '*/10'
#    - hour: '*'
#    - daymonth: '*'
#    - month: '*'
#    - dayweek: '*'
```

and then ran `sudo salt hostname_standalone state.apply`, (where hostname is the actual hostname).  That did not complain but it has not removed the lines from the crontab or stopped the emails.

**Comment by weslambert** [comment]
Most of the  cron tab entries are managed by Salt, so you would need to copy `/opt/so/saltstack/default/salt/zeek.init.sls` to `/opt/so/saltstack/local/salt/zeek/init.sls`, and modify it there to suit your needs for the time being -- we probably need to look at modifying the script/cron job so that it is not emailed.

**Reply by launchdaemon** [reply]
Hi Wes,

Thanks for your reply.  I have tried copying that file and then commented out the following lines:

```
#plcronscript:
#  file.managed:
#    - name: /usr/local/bin/packetloss.sh
#    - source: salt://zeek/cron/packetloss.sh
#    - mode: 755

#zeekpacketlosscron:
#  cron.present:
#    - name: /usr/local/bin/packetloss.sh
#    - user: root
#    - minute: '*/10'
#    - hour: '*'
#    - daymonth: '*'
#    - month: '*'
#    - dayweek: '*'
```
I've then run `sudo salt-call state.highstate` which had a lot of output and included:

```
          ID: socore_own_saltstack
    Function: file.directory
        Name: /opt/so/saltstack
      Result: True
     Comment: Directory /opt/so/saltstack updated
     Started: 15:16:20.681151
    Duration: 980.814 ms
     Changes:   
              ----------
              /opt/so/saltstack/local/salt/zeek/init.sls:
                  ----------
                  group:
                      socore
                  user:
                      socore
              group:
                  socore
              user:
                  socore
```
however checking the crontab with `sudo crontab -u root -e` the lines are still there and I am still receiving the emails.  Should I be applying the state or editing the files differently?

Thanks

**Reply by weslambert** [reply]
You can change the cron.present to cron.absent to remove it, else it will remain.  This script does calculate packet loss for Zeek, which is viewable within Grafana -- be aware that by disabling it you will lose that visibility.  It may be best to modify the script, if possible.  We will need to look into it on our side for a more complete solution in the future.
